cmd/charm/charmcmd/push.go

```diff
@@ -203,9 +202,10 @@ func (c *pushCommand) pushResources(ctxt *cmd.Context, client *csClient, meta *c
 		if err := c.uploadResource(uploadResourceParams{
 			ctxt:         ctxt,
 			client:       client,
+			meta:         meta,
 			charmId:      c.id,
 			resourceName: resourceName,
-			filePath:     filePath,
+			reference:    filePath,
```

Comment:
The above AbsPath call doesn't look quite right to me - we don't know it's a resource of file type yet.